src/binder/bind_node_visitor.cpp

```diff
@@ -362,6 +362,7 @@ void BindNodeVisitor::Visit(parser::InsertStatement *node, parser::ParseResult *
           //  This appears to be a bad assumption. We should rename it to DeriveReturnValueTypeForAggregates()
           //  or else fix up any other codepaths. I've currently fixed it for ConstantValueExpression.
           auto expr = values[i];
+          expr->DeriveReturnValueType();
```

Comment:
@lmwnshn there is also a TODO above here

Reply:
Do we need this call? If we don't, I would remove it for now. It re-walks the entire tree.